From 83927a915215f3b655f4e2140a895d588619270f Mon Sep 17 00:00:00 2001
From: Pavan S <psreenivas@habana.ai>
Date: Tue, 23 May 2023 13:04:30 +0300
Subject: [PATCH] [SW-143472] gaudi2: erom: use recovery-conf in erom

The erom recovery is a mechanism to recover the card
by updating firmware hence it should only load and
execute the fit with load type as flash. Hence the
recovery-conf node is added in agent fw loader to
identify the fits that are supported in erom. This
patch adds support to read the fit configuration from
recovery-conf node and executes the agent from the
fit which contain recovery-conf node.

Signed-off-by: Pavan S <psreenivas@habana.ai>
Change-Id: I70a222e868ff7aa9a4ddb37ab08c77de495d3158
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index f416d2b..7c4baa2 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -29,7 +29,7 @@
 #define FIT_IMAGES_PATH		"/images"
 #define FIT_CONFS_PATH		"/configurations"
 
-#define FIT_CONF_NAME		"preboot-conf"
+#define FIT_CONF_NAME		"recovery-conf"
 #define FIT_DESC_PROP		"description"
 #define FIT_LOADABLE_PROP	"loadables"
 #define FIT_HASHED_NODES_PROP	"hashed-nodes"
@@ -190,7 +190,14 @@
 static int fit_loadables_node_get(const void *fit, int id)
 {
 	int images, conf = fit_find_config_node(fit);
-	const char *uname = fit_loadables_name(fit, conf, id);
+	const char *uname;
+
+	if (conf == -1) {
+		preboot_puts("Wrong FIT type: Cannot find recovery conf\n");
+		return conf;
+	}
+
+	uname = fit_loadables_name(fit, conf, id);
 
 	images = fdt_path_offset(fit, FIT_IMAGES_PATH);
 	return fdt_subnode_offset(fit, images, uname);
